plugin manager: add `--level=[major|minor|patch]` (default: `minor`) (#16899)
elastic/logstash: docs/static/plugin-manager.asciidoc

<1> updates all installed plugins
<2> updates only the plugin you specify

> [NOTE]

Author: Preview: https://logstash_bk_16899.docs-preview.app.elstc.co/guide/en/logstash/master/working-with-plugins.html#updating-plugins

I added this as an admonition because in the normal case users shouldn't need to concern themselves with this and will be happy to get the latest _fully-compatible_ version available. It is rare that a user would _want_ to jump across major versions of a plugin because we are taking care to backport non-breaking features and fixes to the previous minor.

.Major-version plugin updates
==============================
By default, the plugin manager will only update plugins for which newer _minor_ or _patch_ versions exist, to avoid the introduction of breaking changes.
If you wish to also include breaking changes, specify `--level=major`.

[source,shell]
----------------------------------
bin/logstash-plugin update --level=major <1>
bin/logstash-plugin update --level=major logstash-input-github <2>
----------------------------------
<1> updates all installed plugins to latest, including major-version breaking changes
<2> updates only the plugin you specify to latest, including major-version breaking changes

> ==============================

Reviewer: What do you think about making this content a sub-heading? 
<img width="1396" alt="Screenshot 2025-01-17 at 10 07 47 AM" src="https://github.com/user-attachments/assets/38925e37-32fd-4f24-9420-4e60139860e5" />

IMO, it's just as noticeable, but less visually busy.  Also, making this a subhead will boost our SEO performance and findability for this info. 

If you like this approach, I can add working content from my local as a suggestion.  If you prefer the current treatment, let me know and I'll add a few wording tweaks.

Author: @karenzone ah _ha_! I was missing the `[discrete]`, which is why my editor complained about _"section title out of sequence"_ for a four-equals sub-heading. I agree, the sub-heading is less visibly busy.

```suggestion
[discrete]
[[updating-plugins-major-versions]]
==== Major-version plugin updates

By default, the plugin manager will only update plugins for which newer _minor_ or _patch_ versions exist, to avoid the introduction of breaking changes.
If you wish to also include breaking changes, specify `--level=major`.

[source,shell]
----------------------------------
bin/logstash-plugin update --level=major <1>
bin/logstash-plugin update --level=major logstash-input-github <2>
----------------------------------
<1> updates all installed plugins to latest, including major-version breaking changes
<2> updates only the plugin you specify to latest, including major-version breaking changes

```

<1> updates all installed plugins
<2> updates only the plugin you specify

[discrete]
[[updating-plugins-major-versions]]
==== Major-version plugin updates

> By default, the plugin manager will only update plugins for which newer _minor_ or _patch_ versions exist, to avoid the introduction of breaking changes.

Author: ```suggestion
To avoid introducing breaking changes, the plugin manager updates plugins for which newer minor or patch versions exist by default. 
```

> [discrete]
> [[updating-plugins-major-versions]]
> ==== Major-version plugin updates
> 
> To avoid introducing breaking changes, the plugin manager updates plugins for which newer minor or patch versions exist by default. 
> If you wish to also include breaking changes, specify `--level=major`.
> 
> [source,shell]
> ----------------------------------
> bin/logstash-plugin update --level=major <1>
> bin/logstash-plugin update --level=major logstash-input-github <2>
> ----------------------------------
> <1> updates all installed plugins to latest, including major-version breaking changes
> <2> updates only the plugin you specify to latest, including major-version breaking changes

Reviewer: ```suggestion
[discrete]
[[updating-major]]
==== Major version plugin updates

To avoid introducing breaking changes, the plugin manager updates only plugins for which newer _minor_ or _patch_ versions exist by default.
If you wish to also include breaking changes, specify `--level=major`.

[source,shell]
----------------------------------
bin/logstash-plugin update --level=major <1>
bin/logstash-plugin update --level=major logstash-input-github <2>
----------------------------------
<1> updates all installed plugins to latest, including major versions with breaking changes
<2> updates only the plugin you specify to latest, including major versions with breaking changes
```
